### PR TITLE
init commit of trajectoryformer

### PR DESCRIPTION
**Pull request by Cedarch**

### CODE REVIEW

File: efg/data/augmentations3d.py
Comment (poodarchu) [inline efg/data/augmentations3d.py:1]: do not include ad-hoc in library. 
the key "pred" / "future" are specific terms in your impl, so keep these modifications in the experiment directory.

File: cli/data_preparation/waymo/create_data.py [hunk 1]
```diff
@@ -62,35 +76,39 @@ def _fill_infos(root_path, frames, split="train", nsweeps=1):
...
         if split != "test":
             # read boxes
+
             annos = ref_obj["objects"]
             num_points_in_gt = np.array([ann["num_points"] for ann in annos])
             gt_boxes = np.array([ann["box"] for ann in annos]).reshape(-1, 9)
             difficulty = np.array([ann["detection_difficulty_level"] for ann in annos])
-
+            gt_ids = np.array([ann["name"] for ann in annos])
             gt_names = np.array([TYPE_LIST[ann["label"]] for ann in annos])
```
Comment (poodarchu) [inline cli/data_preparation/waymo/create_data.py:111]: Is gt_ids and gt_names the same?

File: cli/data_preparation/waymo/create_data.py [hunk 2]
```diff
@@ -62,35 +76,39 @@ def _fill_infos(root_path, frames, split="train", nsweeps=1):
...
-
         info = {
             "path": lidar_path,
             "anno_path": ref_path,
             "token": frame_name,
             "timestamp": ref_time,
+            "veh_to_global": ref_pose,
             "sweeps": [],
+            "frame_time": int(ref_obj['frame_name'].split("_")[-1]),
+            "context_name": ref_obj['scene_name']
```
Comment (poodarchu) [inline cli/data_preparation/waymo/create_data.py]: make the key name the same as items in ref_obj?

File: cli/data_preparation/waymo/create_data.py [hunk 3]
```diff
@@ -62,35 +76,39 @@ def _fill_infos(root_path, frames, split="train", nsweeps=1):
...
-
+            gt_ids = np.array([ann["name"] for ann in annos])
             gt_names = np.array([TYPE_LIST[ann["label"]] for ann in annos])
             mask_not_zero = (num_points_in_gt > 0).reshape(-1)
 
             # filter boxes without lidar points
             annos_dict = {}
             annos_dict["gt_boxes"] = gt_boxes[mask_not_zero, :].astype(np.float32)
             annos_dict["gt_names"] = gt_names[mask_not_zero].astype(str)
+            annos_dict["gt_ids"] = gt_ids[mask_not_zero]
```
Comment (poodarchu) [inline cli/data_preparation/waymo/create_data.py:118]: gt_ids & gt_names?

File: cli/data_preparation/waymo/create_data.py [hunk 4]
```diff
@@ -102,6 +120,12 @@ def _fill_infos(root_path, frames, split="train", nsweeps=1):
                         "token": frame_name,
                         "transform_matrix": None,
                         "time_lag": 0,
+                        "veh_to_global": ref_obj["veh_to_global"],
+                        "gt_boxes": None,
+                        "gt_names": None,
+                        "difficulty": None,
+                        "name": None,
+                        "num_points_in_gt": None
                     }
```
Comment (poodarchu) [inline cli/data_preparation/waymo/create_data.py:135]: gt_boxes, gt_names, diffuculty, num_points_in_gt should be put in an "annotations" dict same as key-frame. 
L178 has saved gt_boxes as annotations, why put them here again?

**Review by poodarchu — COMMENTED** [review]
Make sure all .py files satisfy the codestyle check in CI.

File: cli/data_preparation/waymo/create_data.py (hunk 1 above)
Comment (Cedarch) [reply cli/data_preparation/waymo/create_data.py:100]: gt_ids is unique tracking_id and gt_names id class names

**Review by poodarchu — COMMENTED** [review]
LGTM

**Review by poodarchu — APPROVED** [review]

**Pull request merged**